src/database/schema/ChatSchema.ts

```diff
@@ -446,6 +472,29 @@ class ChatSchema implements BaseDbSchema {
     }
   };
 
+  static updateDeletedChatType = async (
+    db: SQLiteDatabase,
+    messageId: string,
+    oldChat?: ChatSchema
+  ) => {
+    try {
+      const rawJson = oldChat?.rawJson || {};
+      rawJson.message_type = MESSAGE_TYPE_DELETED;
+      rawJson.text = DELETED_MESSAGE_TEXT;
+      const updatedRawJson = JSON.stringify(rawJson);
+
+      const updateQuery = `UPDATE ${ChatSchema.getTableName()}
+        SET type = ?, message = ?, raw_json = ?, attachment_json = null
+        WHERE id = ?;`;
+
+      const updateReplacement = ['deleted', DELETED_MESSAGE_TEXT, updatedRawJson, messageId];
+
+      await db.executeSql(updateQuery, updateReplacement);
+    } catch (e) {
+      console.log('error updating deleted chat:', e);
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This method updates the chat type to 'deleted' when a message is deleted. However, it doesn't handle the case where `oldChat?.rawJson` is not a valid JSON object. If `oldChat?.rawJson` is a string or null, `rawJson.message_type = MESSAGE_TYPE_DELETED;` will throw an error. You should parse `oldChat?.rawJson` before using it as an object.
```diff
-      const rawJson = oldChat?.rawJson || {};
+      let rawJson;
+      try {
+        rawJson = JSON.parse(oldChat?.rawJson) || {};
+      } catch (e) {
+        console.log('error parsing rawJson:', e);
+        rawJson = {};
+      }
```
Also, consider adding error handling for `JSON.stringify(rawJson)` as it can throw an error if `rawJson` is undefined or contains circular references.


<!-- This is an auto-generated comment by OSS CodeRabbit -->